plugins/modules/aci_l4l7_device.py

```diff
@@ -0,0 +1,352 @@
+        device=dict(type='str', aliases=['device_name',
+                                         'logical_device_name']),
+        state=dict(type='str', default='present',
+                   choices=['absent', 'present', 'query']),
+        context_aware=dict(type='str', choices=['single', 'multi']),
+        dev_type=dict(type='str', choices=['physical', 'virtual']),
+        func_type=dict(type='str', choices=['None',
+                                            'GoTo',
+                                            'GoThrough',
+                                            'L1',
+                                            'L2']),
+        is_copy=dict(type='bool'),
+        managed=dict(type='bool'),
+        prom_mode=dict(type='bool'),
+        svc_type=dict(type='str', choices=['adc',
+                                           'fw',
+                                           'others',
+                                           'copy']),
+        trunking=dict(type='bool'),
+        domain=dict(type='str', aliases=['domain_name']),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ['state', 'absent', ['tenant', 'device']],
+            ['state', 'present', ['tenant', 'device']]
+        ]
+    )
+
+    aci = ACIModule(module)
+
+    tenant = module.params.get('tenant')
+    state = module.params.get('state')
+    device = module.params.get('device')
+    context_aware = module.params.get('context_aware')
+    dev_type = module.params.get('dev_type')
+    func_type = module.params.get('func_type')
+    is_copy = aci.boolean(module.params.get('is_copy'))
+    managed = aci.boolean(module.params.get('managed'))
+    prom_mode = aci.boolean(module.params.get('prom_mode'))
+    svc_type = module.params.get('svc_type')
+    trunking = aci.boolean(module.params.get('trunking'))
+    domain = module.params.get('domain')
+
+    aci.construct_url(
+        root_class=dict(
+            aci_class='fvTenant',
+            aci_rn='tn-{0}'.format(tenant),
+            module_object=tenant,
+            target_filter={'name': tenant},
+        ),
+        subclass_1=dict(
+            aci_class='vnsLDevVip',
+            aci_rn='lDevVip-{0}'.format(device),
+            module_object=device,
+            target_filter={'name': device},
+        ),
+        child_classes=['vnsRsALDevToPhysDomP', 'vnsCDev']
+    )
+
+    aci.get_existing()
+
+    if state == 'present':
+        if domain:
+            tdn = 'uni/phys-{0}'.format(domain)
+        else:
+            tdn = None
+        aci.payload(
+            aci_class='vnsLDevVip',
+            class_config=dict(
+                name=device,
+                contextAware='{0}-Context'.format(context_aware),
+                devtype=dev_type.upper(),
+                funcType=func_type,
+                isCopy=is_copy,
+                managed=managed,
+                mode='legacy-Mode',
```

Comment:
Do we need to specify this in payload? Default value should be the same so don't think it is needed to specify in payload.

Reply:
removed